library/src/ui/index.js

```diff
@@ -1 +1,92 @@
-window.CoNDeT.ui = {}
\ No newline at end of file
+window.CoNDeT.ui = {
+  BaseComponent: {
+    init: function (common, props) {
+      this.ref = this.onInit(common, props);
+      this.common = common;
+    },
+    update: function (common, props) {
+      this.ref.onUpdate(common, props);
+      this.common = common;
+    },
+    destroy: function (common) {
+      if (this.children != null) {
+        for (let i = 0; i < children.length; i++) {
+          children[i].destroy(common);
+        }
+      }
+      this.onDestroy(common);
+    },
+    getPosition: function () {
+      return {
+        x: this.ref.offsetLeft,
+        y: this.ref.offsetTop,
+      };
+    },
+    getDimensions: function () {
+      return {
+        width: this.ref.clientWidth,
+        height: this.ref.clientHeight,
+      };
+    },
+    containsPoint: function (x, y) {
+      return (
+        x >= this.ref.offsetLeft &&
+        x <= this.ref.offsetLeft + this.ref.clientWidth &&
+        y >= this.ref.offsetTop &&
+        y <= this.ref.offsetTop + this.ref.clientHeight
+      );
+    },
+    setState: function (state) {
+      if (this.state == null) return;
+      this.state.onDestroy();
+      this.state = state;
+      this.state.onInit();
```

Comment:
W sumie jeżeli `this.state` jest `undefined` na początku, to zawsze przy wywołaniu tej metody nic się nie stanie. Trzeba jedynie linię 41 wrzucić w if-a o zanegowanym warunku